packages/flutter/lib/src/animation/curves.dart

```diff
@@ -305,6 +305,129 @@ class Cubic extends Curve {
   }
 }
 
+/// A cubic polynomial composed of two curves that share a common center point,
+/// mapping of the unit interval.
+///
+/// The curve runs through three points: (0,0), the middle point ([dx], [dy]),
+/// and (1,1).
+///
+/// The [Curves] class contains a curve defined with this class:
+///
+///  * [Curves.easeInOutCubicEmphasized]
+///
+/// The [ThreePointCubic] class implements third-order Bézier curves, where two
+/// curves share an interior control point that the curve passes through. If the
+/// control points surrounding the middle point are not colinear with the middle
+/// point, then the curve's derivative will have a discontinuity (a cusp) at the
+/// shared middle point.
+///
+/// See also:
+///
+///  * [Curves], where many more predefined curves are available.
+///  * [Cubic], which defines a single cubic polynomial.
+///  * [CatmullRomCurve], a curve which passes through specific values.
+class ThreePointCubic extends Curve {
+  /// Creates two cubic curves that share a common control point.
+  ///
+  /// Rather than creating a new instance, consider using one of the common
+  /// three-point cubic curves in [Curves].
+  ///
+  /// The arguments correspond to the control points for the two curves,
+  /// including the middle join point, but do not including the two implied end
+  /// points at 0,0 and 1,1, which are fixed.
+  const ThreePointCubic(
+      this.adx1, this.ady1, this.adx2, this.ady2,
```

Comment:
Would it make sense to represent the parameters in terms of `Cubic`, `Offset`, `Cubic`?

Reply:
Well, I thought about that, but then the cubics would have to have their scales and offsets already incorporated, so the values wouldn't all be in the same coordinate space, and you'd have to transform them by hand when making them. I could do the scales/offsets in the constructor, but then the constructor wouldn't be a const constructor, and it's actually the same speed as not doing that, once the optimizer is done with it all.

After some benchmarking, I have changed it to take `Offsets` instead of a bunch of doubles, which makes it easier to read/understand, and is about 25% faster on VM runs, and the same speed on AOT runs.